src/catalog/src/local/memory.rs

```diff
@@ -260,6 +245,29 @@ impl CatalogManager for MemoryCatalogManager {
 }
 
 impl MemoryCatalogManager {
+    /// Yes this is the `default()` constructor
+    #[allow(clippy::should_implement_trait)]
+    pub fn default() -> Arc<Self> {
```

Comment:
`default()` should not do too many things, may you consider moving to the another constructor